### PR TITLE
Added function to perform cumulative integrals, and array inputs

### PR DESCRIPTION
**Pull request by nbrantut**
The idea was to be able to compute numerical integrals for each value of input x, simiilar to what matlab does with cumtrapz instead of trapz. This is only relevant for Traezoidal method because higher order methods require vales of integrand at more than 2 points, so the ends would be difficult to deal with.

Also added methods to get integrals for each column when input y in an array.
Also added some minimal documentation for each function.

I am aware that the implementation is not very generic but it seems to work for the intended purpose...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,4 +41,6 @@ The currently available methods are:
 - SimpsonEven
 - SimpsonEvenFast
 
+Only Trapezoidal methods are available for cumumlative integrals.
```
Comment (dextorious) [inline README.md]: Typo: "cumumlative" -> "cumulative".

File: README.md [hunk 2]
```diff
@@ -16,13 +16,21 @@ Do note that while the code is trivial, it has not been extensively tested and d
 ```julia
 # setup some data
 x = collect(-π : π/1000 : π)
-y = sin(x)
+y = sin.(x)
```
Comment (dextorious) [inline README.md:19]: Good catch, thanks.

File: src/NumericalIntegration.jl [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module NumericalIntegration
 
-export integrate
+export integrate, cintegrate
```
Comment (dextorious) [inline src/NumericalIntegration.jl]: I'm tempted to call it `cumulative_integrate` in keeping with the rest of the duck typing in the code base. If you think that's too long, `cintegrate` should be fine.

File: src/NumericalIntegration.jl [hunk 4]
```diff
@@ -16,11 +16,41 @@ struct SimpsonEvenFast     <: IntegrationMethod end
...
+#implementation
+
 function _zero(x,y)
     ret = zero(eltype(x)) + zero(eltype(y))
     ret / 2
 end
 
+function _zeros(x::AbstractVector,y::AbstractVector)
+    ret = zeros(eltype(x),size(x)) + zeros(eltype(y),size(y))
+    ret / 2
```
Comment (dextorious) [inline src/NumericalIntegration.jl]: I'm a bit confused here. What does the `ret / 2` do for zeros?

File: src/NumericalIntegration.jl [hunk 5]
```diff
@@ -71,6 +126,90 @@ function integrate(x::AbstractVector, y::AbstractVector, ::SimpsonEvenFast)
     @inbounds return (x[2] - x[1]) * retval
 end
 
+"""
+    integrate(x::AbstractVector, y::AbstractArray, method)
+
+When y is an array, compute integral for each column.
+"""
+function integrate(x::AbstractVector, y::AbstractArray, M::IntegrationMethod)
```
Comment (dextorious) [inline src/NumericalIntegration.jl]: Why columns in particular? I think this should take an `axis` argument that can default to columns to keep with the MATLAB convention.

Comment (nbrantut) [reply src/NumericalIntegration.jl]: Yep, good point. I can do that. I chose the keyword term "dims" to make it consistent with some of the Base functions (like sum() etc).

File: src/NumericalIntegration.jl (hunk 4 above)
Comment (nbrantut) [reply src/NumericalIntegration.jl]: I guess I just copied-and-pasted waht was in the definition of _zero(x,y) above. I think both lines (41 and 46) can be removed.

File: src/NumericalIntegration.jl (hunk 3 above)
Comment (nbrantut) [reply src/NumericalIntegration.jl]: Fair point: what about cumul_integrate ? It sounds better to me and is still reasonably short.

**Comment by nbrantut** [comment]
Just a suggestion (see recent commit): maybe revert to Trapezoidal() instead of TrapezoidalFast() as the default method ? It is too easy to get caught with wrong dimensions when using the functions for arrays and specifying dimensions.

**Comment by dextorious** [comment]
Seems reasonable to me. Assuming I didn't screw up the merge conflicts just now and the tests pass, I'll merge this tomorrow.

**Pull request merged**